server/libmain/libmain.go

```diff
@@ -182,7 +182,15 @@ func GetConfiguredEnvironmentOrDie(configurator *config.Configurator, healthChec
 		log.Printf("Proxy: forwarding build events to: %s", target)
 	}
 	realEnv.SetBuildEventProxyClients(buildEventProxyClients)
-	realEnv.SetBuildEventHandler(build_event_handler.NewBuildEventHandler(realEnv))
+	beHandler := build_event_handler.NewBuildEventHandler(realEnv)
+	realEnv.SetBuildEventHandler(beHandler)
+
+	sr := beHandler.GetStatsRecorder()
+	sr.Start()
+	realEnv.GetHealthChecker().RegisterShutdownFunction(func(ctx context.Context) error {
+		sr.Stop()
+		return nil
+	})
```

Comment:
you have a reference to realEnv inside of build_event_handler -- would it be cleaner to do this inside of NewBuildEventHandler()?

Reply:
Wasn't sure if this would cause problems for testing, but looks like it works just fine -- done